mobile/lib/modules/login/providers/authentication.provider.dart

```diff
@@ -93,6 +93,7 @@ class AuthenticationNotifier extends StateNotifier<AuthenticationState> {
   Future<bool> logout() async {
     state = state.copyWith(isAuthenticated: false);
```

Comment:
This state change happens before all the cache invalidation etc. If something goes wrong inbetween, eg the logout api call crashes things, will that cause a problem with stale cache?